pkg/machine/sockets.go

```diff
@@ -0,0 +1,87 @@
+package machine
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"net"
+	"path/filepath"
+	"time"
+)
+
+// SetSocket creates a new machine file for the socket and assigns it to
+// `socketLoc`
+func SetSocket(socketLoc *VMFile, path string, symlink *string) error {
+	socket, err := NewMachineFile(path, symlink)
+	if err != nil {
+		return err
+	}
+	*socketLoc = *socket
+	return nil
+}
+
+// ReadySocketPath returns the filepath for the ready socket
+func ReadySocketPath(runtimeDir, machineName string) string {
+	return filepath.Join(runtimeDir, fmt.Sprintf("%s_ready.sock", machineName))
+}
+
+// ListenAndWaitOnSocket waits for a new connection to the listener and sends
+// any error back through the channel. ListenAndWaitOnSocket is intended to be
+// used as a goroutine
+func ListenAndWaitOnSocket(errChan chan<- error, connChan chan<- net.Conn, listener net.Listener) {
```

Comment:
Looks like both caller doesn't even use the returned conn other than directly closing it. So it would be better to not just return it at all and close the conn inside this function here. But I won't block over it.

Reply:
Ah you're right. I will need to test this, but I'm curious if we need to still be connected to the socket in order to successfully start vfkit via `cmd.Start()`. Speaking in terms of AppleHV specifically here of course.

Reply:
Nevermind, I read the code wrong and we call `cmd.Start()` before we actually get the conn from the channel. I can fix this, I would rather get it right the first time rather than go back afterwards.